src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -283,5 +283,44 @@ private static Conversion ToConversion(OverloadResolutionResult<MethodSymbol> re
 
             return new Conversion(ConversionKind.MethodGroup, method, methodGroup.IsExtensionMethodGroup);
         }
+
+        public override Conversion GetStackAllocConversion(BoundStackAllocArrayCreation sourceExpression, TypeSymbol destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        {
+            Debug.Assert((object)sourceExpression.Type == null);
+            Debug.Assert(sourceExpression.ElementType != null);
+
+            var pointerConversion = default(Conversion);
+
+            if (sourceExpression.Syntax.IsVariableDeclarationInitialization())
+            {
+                var sourceAsPointer = new PointerTypeSymbol(sourceExpression.ElementType);
+                pointerConversion = ClassifyImplicitConversionFromType(sourceAsPointer, destination, ref useSiteDiagnostics);
+            }
+
+            if (pointerConversion.IsValid)
+            {
+                // Report unsafe errors
+                _binder.ReportUnsafeIfNotAllowed(sourceExpression.Syntax.Location, ref useSiteDiagnostics);
+
+                return Conversion.MakeStackAllocToPointerType(pointerConversion);
```

Comment:
I think Chuck is suggesting
``` c#
if (...IsVariableDeclarationInitialization())
{
   var pointerConversion = ...;
   if (pointerConversion.IsValid)
   {
       ...
       return ...;
   }
}

var spanType = _binder.GetWellKnownType(WellKnownType.System_Span_T, ref useSiteDiagnostics).Construct(sourceExpression.ElementType);
var spanConversion = ClassifyImplicitConversionFromType(spanType, destination, ref useSiteDiagnostics);
etc
```


---
In reply to: [132780847](https://github.com/dotnet/roslyn/pull/21006#discussion_r132780847) [](ancestors = 132780847,132315521)